systems-analysis/cache-task-support.md

```diff
@@ -0,0 +1,847 @@
+# CacheTask Support in Dragonfly
+
+## Overview
+
+This design document aims to extend the task processing capabilities of the Dragonfly client by introducing a new task type called CacheTask, which enables file distribution based entirely on in-memory caching. Unlike the existing Task type, CacheTask completely bypasses the disk storage layer (Content Storage) and interacts only with the in-memory cache layer (Cache Storage), thereby eliminating disk I/O bottlenecks and significantly improving the distribution performance of hot files.
+
+## Motivation
+
+- **Performance**: Pure in-memory operations eliminate disk I/O overhead, providing faster data access compared to existing cache systems that require disk backup.
+- **Efficiency**: Avoids complex disk garbage collection logic.

```

Comment:
Cache tasks are stored in memory and also need GC. I don't think this is a benefit.